services/provider/server/server_test.go

```diff
@@ -0,0 +1,298 @@
+		Data: map[string]string{
+			"userID":  "cephuser-rbd-provisioner",
+			"userKey": "AQADw/hhqBOcORAAJY3fKIvte++L/zYhASjYPQ==",
+		},
+	},
+	{
+		Name: "rook-ceph-client-cephuser-rbd-node",
+		Kind: "Secret",
+		Data: map[string]string{
+			"userID":  "cephuser-rbd-node",
+			"userKey": "AQADw/hhqBOcORAAJY3fKIvte++L/zYhASjYPQ==",
+		},
+	},
+	{
+		Name: "rook-ceph-client-cephuser-cephfs-provisioner",
+		Kind: "Secret",
+		Data: map[string]string{
+			"adminID":  "cephuser-cephfs-provisioner",
+			"adminKey": "AQADw/hhqBOcORAAJY3fKIvte++L/zYhASjYPQ==",
+		},
+	},
+	{
+		Name: "rook-ceph-client-cephuser-cephfs-node",
+		Kind: "Secret",
+		Data: map[string]string{
+			"adminID":  "cephuser-cephfs-node",
+			"adminKey": "AQADw/hhqBOcORAAJY3fKIvte++L/zYhASjYPQ==",
+		},
+	},
+	{
+		Name: "rook-ceph-client-cephuser-health-checker",
+		Kind: "Secret",
+		Data: map[string]string{
+			"userID":  "cephuser-health-checker",
+			"userKey": "AQADw/hhqBOcORAAJY3fKIvte++L/zYhASjYPQ==",
+		},
+	},
+}
+
+func TestGetJSONConnectionDetails(t *testing.T) {
+	ctx := context.TODO()
+	object := []runtime.Object{
+		&v1.ConfigMap{},
+		&v1.Secret{},
+		&v1.Service{},
+		&rookCephv1.CephClient{},
+	}
+	s := runtime.NewScheme()
+	assert.NoError(t, v1.AddToScheme(s))
+
+	client := newFakeClient(t, object...)
+
+	server := &OCSProviderServer{
+		client:    client,
+		namespace: "openshift-storage",
+	}
+
+	consumerResource := &ocsv1alpha1.StorageConsumer{
+		Status: ocsv1alpha1.StorageConsumerStatus{
+			CephResources: []ocsv1alpha1.CephResourcesSpec{
+				{
+					Name: "cephblockpool",
+					Kind: "CephBlockPool",
+					CephClients: map[string]string{
+						"node":        "cephuser-rbd-node",
+						"provisioner": "cephuser-rbd-provisioner",
+					},
+				},
+				{
+					Name: "cephFilesystemSubVolumeGroup",
+					Kind: "CephFilesystemSubVolumeGroup",
+				},
+				{
+					Name: "cephuser-rbd-provisioner",
+					Kind: "CephClient",
+				},
+				{
+					Name: "cephuser-rbd-node",
+					Kind: "CephClient",
+				},
+				{
+					Name: "cephuser-cephfs-provisioner",
+					Kind: "CephClient",
+				},
+				{
+					Name: "cephuser-cephfs-node",
+					Kind: "CephClient",
+				},
+				{
+					Name: "cephuser-health-checker",
+					Kind: "CephClient",
+				},
+			},
+		},
+	}
+
+	for _, i := range consumerResource.Status.CephResources {
+		if i.Kind == "CephClient" {
+			cephClient, secret := createCephClientAndSecret(i.Name, server)
+			cephClient.Status = &rookCephv1.CephClientStatus{
+				Info: map[string]string{
+					"secretName": fmt.Sprintf("rook-ceph-client-%s", i.Name),
+				},
+			}
+
+			assert.NoError(t, client.Create(ctx, cephClient))
+			assert.NoError(t, client.Create(ctx, secret))
+		}
+	}
+
+	monCm, monSc, mgrSvc := createMonConfigMapandSecret(server)
+	assert.NoError(t, client.Create(ctx, monCm))
+	assert.NoError(t, client.Create(ctx, monSc))
+	assert.NoError(t, client.Create(ctx, mgrSvc))
+
+	exR, err := server.getExternalResources(ctx, consumerResource)
```

Comment:
You should the public entry point (`GetStorageConfig`) not a private util function `getExternalResources`